crypto/src/dex/swap/plaintext.rs

```diff
@@ -38,57 +37,22 @@ impl SwapPlaintext {
         self.claim_address.transmission_key()
     }
 
-    /// Use Blake2b-256 to derive an encryption key `ock` from the OVK and public fields.
-    pub fn derive_ock(ovk: &OutgoingViewingKey, epk: &ka::Public) -> blake2b_simd::Hash {
-        // let cv_bytes: [u8; 32] = cv.into();
-        // let cm_bytes: [u8; 32] = cm.into();
-
-        let mut kdf_params = blake2b_simd::Params::new();
-        kdf_params.hash_length(32);
-        let mut kdf = kdf_params.to_state();
-        kdf.update(&ovk.0);
-        // TODO: should we be using the public fields e.g. t1, t2, trading_pair here?
```

Comment:
Note that by using the new shared `OutgoingCipherKey` logic instead of this method, I'm requiring a note and value commitment, which for swaps I'm assuming would come from the note corresponding to the swap - does this make sense to you @zbuc?

Reply:
I think this fits with the discussions we've had about wanting to ensure that visibility for a `SwapPlaintext` is exactly the same as visibility for the note that records the swap NFT.